SG-21570 beyond you first project (#191)
shotgunsoftware/developer.shotgunsoftware.com: docs/en/guides/pipeline-integrations/administration/beyond-your-first-project.md

  
The configuration, apps and engines are all running on top of the Toolkit Core platform. For new projects, this is also stored inside the  `install` folder. Essentially, a project configuration is fully self contained - all the necessary pieces required to run toolkit are in a single place. This also means that each project is independent and updating one project will not break another.  
  
Tech Notes: Using a shared Toolkit Core (Click to expand)  
  
### Further reading  
  
We also have a more technical document that goes through the high level concepts in the {% include product %} Pipeline Toolkit and explains 'bigger picture' things. Once you have a good grasp of what Toolkit does out of the box, we recommend that you move on to this document to get a deeper undestanding of how Toolkit could be adjusted to suit your particular studio needs.  
  
[An introduction to the high level concepts in the {% include product %} Toolkit](https://support.shotgunsoftware.com/hc/en-us/articles/219040648)  
  

> ## The pipeline Toolkit Community  

Reviewer: ```suggestion
## The Toolkit Community  
```

  
A part of Toolkit is its community of pipeline engineers and TDs! We are on a mission to create a vibrant, code sharing community where we all can help evolve Toolkit together to become a powerful and flexible pipeline environment.  
  
If you have any questions, or want to read through existing posts and conversations, please visit our  [public forums section](https://support.shotgunsoftware.com/hc/en-us/community/topics/200682428-Pipeline-Toolkit-Common-Questions-and-Answers).  
  

> ## Our Support Team  
>   
> Lastly, if you have any questions, ideas or concerns, never hesitate to contact us directly! We try be be practical and hands on with support and if you for example have found a bug, we can often push a bug fix update quickly to you via our app store. Just drop us a line on  `support@shotgunsoftware.com` and we'll be happy to help!

Reviewer: ```suggestion
```

lang: en
---

  
# Beyond your first project  
  
Here, we explain where to go once you have got your first project up and running using the {% include product %} Desktop. It covers useful common questions and topics and lists useful documentation resources.    
  

> _Please note that this document describes functionality only available if you have taken control over a Toolkit configuration. For details, see the  [{% include product %} Integrations User Guide](https://support.shotgunsoftware.com/hc/en-us/articles/115000068574-Integrations-User-Guide#Getting%20Started%20with%20Shotgun%20Desktop)._

Reviewer: ```suggestion
```

  
![](images/Beyond-your-first-project/failed_to_launch.png)  
  
In the toolkit project configuration, we store paths to the various executables that you can launch. If you are seeing the above message, it probably means that those paths are not matching your studio setup. You may also find that the wrong version of the application is being launched; for example, our default configuration may have a path to maya 2015 but your studio is running maya 2014. In this case, you also need to change the paths.  
  
In our default configurations, these paths are all stored in a single file called  `paths.yml`. In order to change a path, locate your project configuration on disk and then navigate into the config folder until you find the  `paths.yml` file:  
  
![](images/Beyond-your-first-project/paths_config.png)  
  
Open this file and make the necessary changes to the paths. Once you have saved the file, you need to leave the project inside of {% include product %} desktop and then click back into it. (but no need to restart the entire application).  
  
**Further Reading**  
  
For more information about applications, check out the following topics:  
   

> [![](https://sg-toolkit.s3.amazonaws.com/tank/docs/apps/tk-multi-launchapp/v0.10.2_1029_133142/icon_256.png)](https://support.shotgunsoftware.com/hc/en-us/articles/219032968) [> The toolkit application launcher](https://support.shotgunsoftware.com/hc/en-us/articles/219032968)  
> 
> [![](https://sg-toolkit.s3.amazonaws.com/tank/docs/apps/tk-multi-launchapp/v0.10.2_1029_133142/icon_256.png)](https://support.shotgunsoftware.com/hc/en-us/articles/219032968#Use%20Command%20Line%20Arguments%20at%20Launch) [> Passing Command line Arguments](https://support.shotgunsoftware.com/hc/en-us/articles/219032968#Use%20Command%20Line%20Arguments%20at%20Launch)    

Reviewer: ```suggestion
- [The Toolkit Application Launcher](https://support.shotgunsoftware.com/hc/en-us/articles/219032968)  
- [Passing Commandline Arguments](https://support.shotgunsoftware.com/hc/en-us/articles/219032968#Use%20Command%20Line%20Arguments%20at%20Launch)    
```

  
# Next Steps  
  
Hopefully at this point you now have the default {% include product %} setup working for a {% include product %} project (or test project). Applications are launching, Context menu actions and publishes are showing up in {% include product %} and things are working on all your desired operating system platforms.  
  
This next section is all about what to do next -- the process of starting to take that default configuration and adjust it to work more like the rest of your studio pipeline. Toolkit is flexible and highly configurable, and we have lots of documentation. But before you get started, to see it all in action, we recommend spending a couple of minutes checking out our various walkthrough videos. These show the {% include product %} Pipeline Toolkit in action, how it works inside applications such as Maya and Nuke. It also goes through basic concepts such as publishing, version control, loading etc.  
  

> [![](https://sg-toolkit.s3.amazonaws.com/tank/docs/docs/17/no_version_0530_193825/icon_256.png)](https://support.shotgunsoftware.com/hc/en-us/articles/219040678) [> {% include product %} Pipeline Toolkit video collection](https://support.shotgunsoftware.com/hc/en-us/articles/219040678)  

Reviewer: ```suggestion
[{% include product %} Toolkit Video Collection](https://support.shotgunsoftware.com/hc/en-us/articles/219040678)  
```

  
In the following sections we'll walk through the various parts of the project configuration folder.  
  
### Command line access  
  
As well as using the {% include product %} Desktop, you can also access Toolkit via a terminal or shell. Each project that you create on disk comes with a special  `tank` command which gives you command line based access to a lot of functionality, including starting up an API session and launching applications.  
  
If you navigate to your project configuration, you can see a  `tank` and a  `tank.bat` command in the root of the configuration. Running these commands without any options will give you a list of all the commands that are supported in your current configuration, including the following useful commands:  
  
- `tank shell` - Start an interactive python shell with tk api access  
- `tank core` - Check if there are any core API updates available for this project  
- `tank updates` - Check if any of the apps or engines in this configuration has got any updates available  
  
For more details on what you can do with the  `tank` command, please see the in-depth technical documentation:  
  

> [![](https://sg-toolkit.s3.amazonaws.com/tank/docs/docs/11/no_version_0530_204710/icon_256.png)](https://support.shotgunsoftware.com/hc/en-us/articles/219033178) [> How to administer Toolkit.](https://support.shotgunsoftware.com/hc/en-us/articles/219033178)  

Reviewer: ```suggestion
[How to Administer Toolkit](https://support.shotgunsoftware.com/hc/en-us/articles/219033178)  
```

  
### Key Configuration Files  
  
The  `config` folder contains a couple of key configuration files.  
  
![](images/Beyond-your-first-project/config_overview.png)  
  
Toolkit comes with a folder creation system which tries to automatically create folders on disk to make sure that when you start up an application, all the necessary structure on disk exists and has been prepared on beforehand! The configuration for this can be found in the  `schema` folder indicated above.  
  
Hand in hand with this goes the Toolkit  _template system_ which makes it easy to define the various paths to files that you can configure; your publishes, work files, renders etc. This is stored in the  `templates.yml` file above.  
  

> Together, these two parts of the project configuration makes it possible to adjust the various Apps that toolkit use to write out data to locations on disk which make sense are are understood by your existing pipeline. Read more about this in our advanced documentation:  
> 
> [![](https://sg-toolkit.s3.amazonaws.com/tank/docs/docs/11/no_version_0530_204710/icon_256.png)](https://support.shotgunsoftware.com/hc/en-us/articles/219033178#Creating%20folders%20on%20disk%20with%20Sgtk) [> Folder Configuration](https://support.shotgunsoftware.com/hc/en-us/articles/219033178#Creating%20folders%20on%20disk%20with%20Sgtk)  
> 
> [![](https://sg-toolkit.s3.amazonaws.com/tank/docs/docs/11/no_version_0530_204710/icon_256.png)](https://support.shotgunsoftware.com/hc/en-us/articles/219033178#Configuring%20Templates) [> File system templates](https://support.shotgunsoftware.com/hc/en-us/articles/219033178#Configuring%20Templates)  

Reviewer: ```suggestion
Together, these two parts of the project configuration makes it possible to adjust the various Apps that toolkit use to write out data to locations on disk which make sense are are understood by your existing pipeline.

Read more about this in our advanced documentation:  

- [Folder Configuration](https://support.shotgunsoftware.com/hc/en-us/articles/219033178#Creating%20folders%20on%20disk%20with%20Sgtk)  
- [Filesystem Templates](https://support.shotgunsoftware.com/hc/en-us/articles/219033178#Configuring%20Templates)  
```

  
A toolkit configuration is essentially made up of a collection of configured  **apps and engines**. This configuration is located in the  `env` folder. If the file system configuration files discussed above define  _where_ resources should be located on disk, the environment configuration with its apps and engines define  _what_ the pipeline is supposed to do.  

> [![](https://sg-toolkit.s3.amazonaws.com/tank/zendesk_appstore/tank_logo_256.png)](https://support.shotgunsoftware.com/hc/en-us/articles/219039798) [> The official Toolkit App and Engine listing.](https://support.shotgunsoftware.com/hc/en-us/articles/219039798)  

Reviewer: ```suggestion
```

  
The configuration, apps and engines are all running on top of the Toolkit Core platform. For new projects, this is also stored inside the  `install` folder. Essentially, a project configuration is fully self contained - all the necessary pieces required to run toolkit are in a single place. This also means that each project is independent and updating one project will not break another.  
  
Tech Notes: Using a shared Toolkit Core (Click to expand)  
  
### Further reading  
  
We also have a more technical document that goes through the high level concepts in the {% include product %} Pipeline Toolkit and explains 'bigger picture' things. Once you have a good grasp of what Toolkit does out of the box, we recommend that you move on to this document to get a deeper undestanding of how Toolkit could be adjusted to suit your particular studio needs.  
  

> [![](https://sg-toolkit.s3.amazonaws.com/tank/docs/docs/10/no_version_0330_043251/icon_256.png)](https://support.shotgunsoftware.com/hc/en-us/articles/219040648) [> An introduction to the high level concepts in the Shotgun Pipeline Toolkit.](https://support.shotgunsoftware.com/hc/en-us/articles/219040648)  

Reviewer: ```suggestion
[An introduction to the high level concepts in the {% include product %} Toolkit](https://support.shotgunsoftware.com/hc/en-us/articles/219040648)  
```

  
_Please note that this document describes functionality only available if you have taken control over a Toolkit configuration. For details, see the  [{% include product %} Integrations User Guide](https://support.shotgunsoftware.com/hc/en-us/articles/115000068574-Integrations-User-Guide#Getting%20Started%20with%20Shotgun%20Desktop)._  
  
#### Table of Contents:  
  
- [Welcome to Toolkit](#welcome-to-toolkit)  
  
- [Basic Configuration](#basic-configuration)  
  
  * [Setting up Application Paths](#setting-up-application-paths)  
  
  * [{% include product %} Integration](#shotgun-integration)  
  
  * [Adding Publishes to the {% include product %} UI](#adding-publishes-to-the-shotgun-ui)  
  
  * [Multiple Operating Systems](#multiple-operating-systems)  
  
- [Next Steps](#next-steps)  
  
  * [The anatomy of a Toolkit Project](#the-anatomy-of-a-toolkit-project)  
  
      + [Command line access](#command-line-access)  
  
      + [Key Configuration Files](#key-configuration-files)  
  
      + [Core API platform](#core-api-platform)  
  
      + [Further reading](#further-reading)  
  
  * [The pipeline Toolkit Community](#the-pipeline-toolkit-community)  
  

>   * [Our Support Team](#our-support-team)  

Reviewer: Let's remove the table of contents entirely. What you have here actually looks pretty good, but I don't think the document is long enough to really need one explicitly built into the document like this.

  
# Welcome to Toolkit  
  
Welcome to Toolkit! If you are reading this, it probably means that you have managed to successfully install your first {% include product %} Pipeline Toolkit Project using the {% include product %} Desktop.  
  

> ![](https://sg-toolkit.s3.amazonaws.com/tank/docs/docs/55/no_version_0513_171143/beyond_your_first_project/project_ready.png)  

Reviewer: Already mentioned this to you elsewhere, but let's get the images into the repo and reference them directly. I'd prefer we stay away from external links like this where possible.
